src/v/serde/avro/tests/data_generator.cc

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright 2024 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+#include "serde/avro/tests/data_generator.h"
+
+#include "base/seastarx.h"
+#include "random/generators.h"
+
+#include <seastar/util/defer.hh>
+
+namespace testing {
+
+::avro::GenericDatum generate_datum(
```

Comment:
q: is this a straight rip and move aside from the function rename?

Reply:
Just a rename + move (sorry I should've split this up, but also I ended up not using this yet and will probably remove the commit)

Reply:
np, just checking to avoid reading the diff

Reply:
This commit is now removed